pkg/operator/staticpodoperator/controlloops/installer_controller.go

```diff
@@ -0,0 +1,377 @@
+		}
+		glog.Infof("%q needs to deploy to %d", currKubeletState.NodeName, deploymentIDToStart)
+
+		// we need to start a deployment create a pod that will lay down the static pod resources
+		newCurrKubeletState, err := c.createInstallerPod(currKubeletState, operatorSpec, deploymentIDToStart)
+		if err != nil {
+			return true, err
+		}
+		// if we make a change to this status, we want to write it out to the API before we commence work on the next kubelet.
+		// it's an extra write/read, but it makes the state debuggable from outside this process
+		if !equality.Semantic.DeepEqual(newCurrKubeletState, currKubeletState) {
+			glog.Infof("%q moving to %v", currKubeletState.NodeName, spew.Sdump(*newCurrKubeletState))
+			operatorStatus.TargetKubeletStates[i] = *newCurrKubeletState
+			if !reflect.DeepEqual(originalOperatorStatus, operatorStatus) {
+				_, updateError := c.operatorConfigClient.UpdateStatus(resourceVersion, operatorStatus)
+				return false, updateError
+			}
+		}
+		break
+	}
+
+	v1alpha1helpers.SetOperatorCondition(&operatorStatus.Conditions, operatorv1alpha1.OperatorCondition{
+		Type:   "InstallerControllerFailing",
+		Status: operatorv1alpha1.ConditionFalse,
+	})
+	if !reflect.DeepEqual(originalOperatorStatus, operatorStatus) {
+		_, updateError := c.operatorConfigClient.UpdateStatus(resourceVersion, operatorStatus)
+		if updateError != nil {
+			return true, updateError
+		}
+	}
+
+	return false, nil
+}
+
+// newKubeletStateForInstallInProgress returns the new KubeletState or error
+func (c *InstallerController) newKubeletStateForInstallInProgress(currKubeletState *KubeletState) (*KubeletState, error) {
+	ret := currKubeletState.DeepCopy()
+	installerPod, err := c.kubeClient.CoreV1().Pods(c.targetNamespace).Get(getInstallerPodName(currKubeletState.TargetDeploymentID, currKubeletState.NodeName), metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		ret.LastFailedDeploymentID = currKubeletState.TargetDeploymentID
+		ret.TargetDeploymentID = currKubeletState.CurrentDeploymentID
+		ret.Errors = []string{err.Error()}
+		return ret, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	switch installerPod.Status.Phase {
+	case corev1.PodSucceeded:
+		ret.CurrentDeploymentID = currKubeletState.TargetDeploymentID
+		ret.TargetDeploymentID = 0
+		ret.LastFailedDeploymentID = 0
+		ret.Errors = nil
+	case corev1.PodFailed:
+		ret.LastFailedDeploymentID = currKubeletState.TargetDeploymentID
+		ret.TargetDeploymentID = 0
+
+		errors := []string{}
+		for _, containerStatus := range installerPod.Status.ContainerStatuses {
+			if containerStatus.State.Terminated != nil && len(containerStatus.State.Terminated.Message) > 0 {
+				errors = append(errors, fmt.Sprintf("%q: %s", containerStatus.Name, containerStatus.State.Terminated.Message))
+			}
+		}
+		if len(errors) == 0 {
+			errors = append(errors, "no detailed termination message, see `oc get -n %q pods/%q -oyaml`", installerPod.Namespace, installerPod.Name)
+		}
+		ret.Errors = errors
+	}
```

Comment:
how about other states? do we care?

Reply:
> how about other states? do we care?

I don't think so.  it just means we haven't run yet.  Perhaps a timeout in the future I suppose.